tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -0,0 +1,756 @@
+
+	storageProfile = &compute.StorageProfile{
+		ImageReference: &compute.ImageReference{
+			Publisher: to.StringPtr(stringSplit[0]),
+			Offer:     to.StringPtr(stringSplit[1]),
+			Sku:       to.StringPtr(stringSplit[2]),
+			Version:   to.StringPtr(stringSplit[3]),
+		},
+	}
+	return storageProfile
+}
+
+// randonPasswordString generates random string with restrictions of given length.
+func randomPasswordString(n int) string {
+	digits := "0123456789"
+	specials := "~=+%^*/()[]{}/!@#$?|"
+	var letter = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	b := make([]byte, n)
+	b[0] = digits[rand.Intn(len(digits))]
+	b[1] = specials[rand.Intn(len(specials))]
+	for i := 2; i < n; i++ {
+		b[i] = letter[rand.Intn(len(letter))]
+	}
+	rand.Shuffle(len(b), func(i, j int) {
+		b[i], b[j] = b[j], b[i]
+	})
+	return string(b)
+}
+
+//randomString generates random string of given length.
+func randomString(n int) string {
+	var letter = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = letter[rand.Intn(len(letter))]
+	}
+	return string(b)
+}
+
+// generateResourceName generates the names for the individual resource components of an instance
+func (az *AzureProvider) generateResourceName(resource, randomStr string) (name string) {
+	name = strings.Join([]string{az.infraID, "windows", "worker", randomStr, resource}, "-")
+	return name
+}
+
+// getIPAdress gets the IP Address by IP resource name as an argument.
+func (az *AzureProvider) getIPAddress(ctx context.Context) (ipAddress string, err error) {
+	result, err := az.ipClient.Get(ctx, az.resourceGroupName, az.IpName, "")
+	if err != nil {
+		return "", err
+	}
+	ipAddressProperties := *(result.PublicIPAddressPropertiesFormat)
+	ipAddress = *(ipAddressProperties.IPAddress)
+	return
+}
+
+// constructWinRMConfig constructs the winrm Listeners for the instance for now it will configure only one listener, but this may
+// scale to multiple listeners depending on the use case. By default they listen via Http protocol.
+func constructWinRMConfig() (winRMConfig *compute.WinRMConfiguration) {
+	winRMListeners := &[]compute.WinRMListener{
+		{
+			Protocol: "Http",
+		},
+	}
+	winRMConfig = &compute.WinRMConfiguration{
+		Listeners: winRMListeners,
+	}
+	return winRMConfig
+}
+
+// constructOSProfile constructs the OS Profile for the creation of windows instance. The OS Profile consists of information
+// such as configuring remote management listeners, instance access setup.
+func (az *AzureProvider) constructOSProfile(ctx context.Context) (osProfile *compute.OSProfile, vmName, password string) {
+	instanceName := "windows-" + randomString(5)
+	adminUserName := "windows"
+	adminPassword := randomPasswordString(12)
+	computeWinRMConfig := constructWinRMConfig()
+	nodeLocation := *(az.getvnetProfile(ctx).Location)
+	var TimeZone string
+	if nodeLocation == "centralus" {
+		TimeZone = "Central Standard Time"
```

Comment:
Please experiment and find out. Otherwise you have to create a mapping for every region that Azure supports for completeness either in this PR or in the future.